### ISSUE TITLE
[bug] Mounting `ticket.db` without `certs.db` can crash in a specific script run scenario in v1.9.0.

### ISSUE THREAD

**Issue by TurdPooCharger** [d0k3/GodMode9#555]
**TL;DR**
Searching for tickets `*.tik` in an image mounted `ticket.db` with nonexistent **[T:] TICKET.DB IMAGE** drive will crash GodMode9 v19.0. 

**Edit** -  Slight update. No need to mount `ticket.db` or relocate `certs.db`. This alone is enough to trigger the crash so long as **[T:] TICKET.DB IMAGE** is not present.

```
if find -f "T:/eshop/*.tik" NULL
end
```

***

**Procedure**

1. Place the script `mount_ticket_db.gm9` at **sdmc:/gm9/scripts**. See the code below -or- download here: [mount_ticket_db.gm9.txt](https://github.com/d0k3/GodMode9/files/3765864/mount_ticket_db.gm9.txt)
     - (Requires removing the `.txt` extension)
2. GodMode9 → **[1:] SYSNAND CTRNAND/dbs/**`certs.db`
3. Press (Y) on `certs.db` to **[CLIPBOARD]** select the file.
4. Back out once with (B) button. You should now be in **[1:] SYSNAND CTRNAND**.
5. Press (Y) → **Move path(s)** → **<A> yes** → button combo → **<A> yes**.
     - The `certs.db` cannot be placed in the same folder as `ticket.db`.
6. Press and hold (START) to exit, return to Luma3DS chainloader, and boot back to GM9.
     - If you do not reset the GM9 session, it might be possible to still access the **[T:] TICKET.DB IMAGE**.  
7. Press (HOME) → **Scripts...** → **mount_ticket_db** → etc.
     - There's no way to manually search for files in **[T:]** drive as it is missing from user mode. The crash can only be replicated in script.

***

**Results**
GM9 v1.8.0 - Script completes its run with this message.
> 1:/dbs/ticket.db
> Mounting image: failed 

GM9 v1.9.0 - Crash happens at line 10:
- `if find -f "T:/eshop/*.tik" NULL`

[exception_dump_010103150318.txt](https://github.com/d0k3/GodMode9/files/3765794/exception_dump_010103150318.txt)

<img width="225" alt="mount ticket db crash" src="https://user-images.githubusercontent.com/35666439/67454970-1685b000-f5fa-11e9-847b-cb9ddb240ce3.png">

***

`mount_ticket_db.gm9`

```
echo "Testing if 'ticket.db' can mount..."

if findnot "1:/dbs/ticket.db" NULL
	echo "1:/dbs/ticket.db\nFile not found."
	goto EXIT
end

imgumount

if imgmount "1:/dbs/ticket.db"
end

  if find -f "T:/eshop/*.tik" NULL
	imgumount
	echo "1:/dbs/ticket.db\nMounting image: success"
elif find -f "T:/hidden/*.tik" NULL
	imgumount
	echo "1:/dbs/ticket.db\nMounting image: success"
elif find -f "T:/system/*.tik" NULL
	imgumount
	echo "1:/dbs/ticket.db\nMounting image: success"
elif find -f "T:/hidden/*.tik" NULL
	imgumount
	echo "1:/dbs/ticket.db\nMounting image: success"
else
	imgumount
	echo "1:/dbs/ticket.db\nMounting image: failed"
end

imgumount

@EXIT
```

***

**Info about your system**
* **Console type:** n3DSXL
* **Anything special about your console?** galaxy edition, purple? Still works somehow.
* **Bootloader:** boot9strap v1.3
* **Did you chainload GodMode9 via Luma?:** Yes

**Have you actually read this?**
[x] I have read the information above

**Comment by aspargas2** [comment]
I'll look into this; it's probably just a simple mistake, but it's the more reason to just do away with the old vtickdb code entirely, lol.

**Comment by d0k3** [comment]
@TurdPooCharger - if I got that right, the only way to trigger this is via a script?

@aspargas2 - looks likt this will be fixed anyway by your new developments, I guess.

**Comment by aspargas2** [comment]
Yeah, moving to full DBRI support should fix this anyway, but it.would still be nice to know what's causing this so I don't make a similar mistake. I'll try to look into it either tomorrow or Sunday. School hadn't been leaving much time for dev stuff recently.

**Comment by Wolfvak** [comment]
Okay, just checked and it seems to be a NULL deref on vtickdb.c:162.

@TurdPooCharger can you test this FIRM? It has a simple NULL-check at the start of `ReadVTickDbDir`, but I'm not too sure if this results in the expected behavior.

[GodMode9-v1.9.0-5-gc38c99e5-20191025080426.zip](https://github.com/d0k3/GodMode9/files/3771731/GodMode9-v1.9.0-5-gc38c99e5-20191025080426.zip)

**Comment by TurdPooCharger** [comment]
@Wolfvak, two additional scripts that were tested to fail in v1.9.0 stable release were also pitted against your hourly build. At the time of testing, my n3DSXL did not have EmuNAND, bonus drive, or mounted nand `*.bin` images. In the second script found below, two fictitious drives were included to see how it handles stupid inputs.

GM9 v1.9.0-5-gc38c99e5 passes with flying colors for all three scripts.

***

`mount_ticket_db.gm9` - **PASSED**

***

`attempt_copy_tickets.gm9` - **PASSED**

```
if findnot "$[GM9OUT]/tickets"
	mkdir "$[GM9OUT]/tickets"
end

if imgmount "1:/dbs/ticket.db"
	cp -o -s -n -w "T:/eshop" "$[GM9OUT]/tickets"
	cp -o -s -n -w "T:/hidden" "$[GM9OUT]/tickets"
	cp -o -s -n -w "T:/system" "$[GM9OUT]/tickets"
	cp -o -s -n -w "T:/unknown" "$[GM9OUT]/tickets"
end
```

***

`search stress test.gm9` - **PASSED**

```
if find -f "0:/luma/payloads/*.firm" NULL
end

echo "0 - SD CARD"

########################

if find -f "1:/dbs/*.db" NULL
end

echo "1 - SYSNAND CTRNAND"

########################

if find -f "2:/sys/*.dat" NULL
end

echo "2 - SYSNAND TWLN"

########################

if find -f "3:/photo/private/ds/app/48E494A/*.bin" NULL
end

echo "3 - SYSNAND TWLP"

########################

if find -f "4:/dbs/*.db" NULL
end

echo "4 - EMUNAND CTRNAND"

########################

if find -f "5:/sys/*.dat" NULL
end

echo "5 - EMUNAND TWLN"

########################

if find -f "6:/photo/private/ds/app/48E494A/*.bin" NULL
end

echo "6 - EMUNAND TWLP"

########################

if find -f "7:dbs/*.db" NULL
end

echo "7 - FAT IMAGE / IMGNAND CTRNAND"

########################

if find -f "8:/sys/*.dat" NULL
end

echo "8 - BONUS DRIVE / IMGNAND TWLN"

########################

if find -f "9:/photo/private/ds/app/48E494A/*.bin" NULL
end

echo "9 - RAM DRIVE / IMGNAND TWLP"

########################

if find -f "A:/dbs/*.db" NULL
end

echo "A - SYSNAND SD"

########################

if find -f "B:/dbs/*.db" NULL
end

echo "B - EMUNAND SD"

########################

if find -f "C:/*.3ds" NULL
end

echo "C - GAMECART"

########################

if find -f "E:/*.bin" NULL
end

echo "E - EMUNAND VIRTUAL"

########################

if find -f "G:/*.bin" NULL
end

echo "G - GAME IMAGE"

########################

if find -f "I:/*.bin" NULL
end

echo "I - IMGNAND VIRTUAL"

########################

if find -f "K:/*.bin" NULL
end

echo "K - AESKEYDB IMAGE"

########################

if find -f "M:/*.mem" NULL
end

echo "M - MEMORY VIRTUAL"

########################

if find -f "S:/*.bin" NULL
end

echo "S - SYSNAND VIRTUAL"

########################

goto SKIP_1

if find -f "T:/eshop/*.tik" NULL
end

echo "T - TICKET.DB IMAGE"

@SKIP_1

########################

if find -f "V:/*.bin" NULL
end

echo "V - VRAM VIRTUAL"

########################

if find -f "Z:/*" NULL
end

echo "Z - LAST SEARCH"

########################
########################

if find -f "10:/*.bin" NULL
end

echo "10 - does not exist"

########################

if find -f "D:/*.bin" NULL
end

echo "D - DERP VIRTUAL"

```

**Comment by Wolfvak** [comment]
Great, I'll push this later today
